### ISSUE TITLE
Have you considered a super large graph(10 million nodes)?

### ISSUE THREAD

**Issue by YijianLiu** [alga-hopf/drl-graph-partitioning#1]
Thanks for your amazing work, and I have 2 questions really need your help.
1. The time you need to achieve partition using your method compared to METIS?
2. Whether this method can be used in a super large graph(10 million nodes), as far as I known, METIS is usuallly used in a large graph(10 million nodes) when training GNN, I don't know whether your method is better than METIS on a super large graph.
Thank you very much and need your help!

**Comment by alga-hopf** [comment]
Thanks for your questions. The code in this repo is ok for graphs up tp 90K node, since there are some bottlenecks that get more severe with larger graphs. For bigger graphs you may want to have a look at this other work https://epubs.siam.org/doi/10.1137/1.9781611977141.3 (here is the repo https://github.com/alga-hopf/dl-spectral-graph-partitioning). The models that we present there are overall faster than METIS.

**Comment by qlinsey** [comment]
Hi, Same question here. 
We clone the code and it is fast to run drl_partitioning_coarsest_train.py. However, when run drl_partitioning_train.py, it has been for one entire day and we have not got results. Do you mind to share with us how long it takes to run drl_partitioning_train.py ( we directly cloned from this Repo)? 

Thanks and hope to get your response soon!

**Comment by alga-hopf** [comment]
Hi! It turned out that there was an issue with multiprocessing during the training. It should be fixed now and much faster. Thanks for pointing this out!

**Comment by qlinsey** [comment]
Hi, Thank you so much! once updated, could you please leave message here so that we will have a sync on our side. Thank you!

**Comment by qlinsey** [comment]
Thank you! we will try again

**Comment by alga-hopf** [comment]
Hi, the code is already updated.

**Comment by qlinsey** [comment]
Yes! I noticed ,Thank you!

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by qlinsey** [comment]
Hi, again here, 

After pull latest code, got exception:  AttributeError: 'Graph' object has no attribute 'node'.  looks like because of networkx version, i re-installed networkx==2.3, still have same exception. May i know your networkx version? thank you.

**Comment by alga-hopf** [comment]
Yes, that might have changed in various networkx versions. Mine is 2.8, hopefully using this version should solve the problem.

**Comment by qlinsey** [comment]
Thank you!

**Comment by qlinsey** [comment]
Hi, thanks for all these response.. May I ask one more question again: 
for "Sparse Matrices from the SuiteSparse Collection", which exact data are you using? I downloaded  odepb400.tar.gz from https://sparse.tamu.edu/, which has 400*400 (row*column), there is a condition: nx.number_connected_components(gnx) == 1  for creating dataloader, but this odepb400 return 400!=1, do you mind to share the dataname which you used for experiment? thanks a lot!
